sqldelight-core/src/test/kotlin/com/squareup/sqldelight/core/queries/MutatorQueryFunctionTest.kt

```diff
@@ -0,0 +1,56 @@
+package com.squareup.sqldelight.core.queries
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.sqldelight.core.compiler.MutatorQueryGenerator
+import com.squareup.sqldelight.core.compiler.namedMutators
+import com.squareup.sqldelight.core.util.FixtureCompiler
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+class MutatorQueryFunctionTest {
+  @get:Rule val tempFolder = TemporaryFolder()
+
+  @Test fun `mutator method generates proper method signature`() {
+    val file = FixtureCompiler.parseSql("""
+      |CREATE TABLE data (
+      |  _id INTEGER NOT NULL PRIMARY KEY,
+      |  value TEXT AS kotlin.collections.List
+      |);
+      |
+      |insertData:
+      |INSERT INTO data
+      |VALUES (?, ?);
+      """.trimMargin(), tempFolder)
+
+    val generator = MutatorQueryGenerator(file.sqliteStatements().namedMutators().first())
+
+    assertThat(generator.function().toString()).isEqualTo("""
+      |fun insertData(_id: kotlin.Long, value: kotlin.collections.List?): kotlin.Long = insertData.execute(_id, value)
+      |""".trimMargin())
+  }
+
+  @Test fun `mutator method generates proper private value`() {
+     val file = FixtureCompiler.parseSql("""
+      |CREATE TABLE data (
+      |  _id INTEGER NOT NULL PRIMARY KEY,
+      |  value TEXT AS kotlin.collections.List
+      |);
+      |
+      |insertData:
+      |INSERT INTO data
+      |VALUES (?, ?);
+      """.trimMargin(), tempFolder)
+
+    val generator = MutatorQueryGenerator(file.sqliteStatements().namedMutators().first())
+
+    assertThat(generator.value().toString()).isEqualTo("""
+      |private val insertData: InsertData by lazy {
```

Comment:
idk how to massage the kotlinpoet code to make this not look like trash. Might be a bug in kotlinpoet.